go.mod

```diff
@@ -1,31 +1,164 @@
 module sigs.k8s.io/boskos
 
-go 1.15
+go 1.17
 
 replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v14.2.0+incompatible
 	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
-	k8s.io/client-go => k8s.io/client-go v0.21.1
+	k8s.io/client-go => k8s.io/client-go v0.23.5
```

Comment:
seems like the prow job is still pointing to older version of golang and k8s, lets rollback version to older version of k8s bits.